dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ranger/RangerClient.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @JsonProperty
+    abstract ImmutableList<Service> services();
+  }
+
+  @AutoValue
+  abstract static class ListPoliciesResponse {
+
+    @JsonCreator
+    static ListPoliciesResponse create(@JsonProperty("policies") ImmutableList<Policy> policies) {
+      return new AutoValue_RangerClient_ListPoliciesResponse(policies);
+    }
+
+    @JsonProperty
+    abstract ImmutableList<Policy> policies();
+  }
+
+  public static class RangerException extends Exception {
+
+    public RangerException(String message, Throwable cause) {
+      super(message, cause);
+    }
+
+    public RangerException(String message) {
+      super(message);
+    }
+  }
+
+  private final URI baseUri;
+
+  private final CloseableHttpClient httpClient;
+
+  private final UsernamePasswordCredentials credentials;
+
+  public RangerClient(URI baseUri, String user, String password) throws URISyntaxException {
+    this.baseUri = baseUri;
+    httpClient = HttpClients.createMinimal();
+    credentials = new UsernamePasswordCredentials(user, password);
+  }
+
+  public ImmutableList<User> findUsers(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/xusers/users", args, ListUsersResponse.class).vXUsers();
+  }
+
+  public ImmutableList<Group> findGroups(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/xusers/groups", args, ListGroupsResponse.class).vXGroups();
+  }
+
+  public ImmutableList<Role> findRoles(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/roles/roles", args, ListRolesResponse.class).roles();
+  }
+
+  public ImmutableList<Service> findServices(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/plugins/services", args, ListServicesResponse.class).services();
+  }
+
+  public ImmutableList<Policy> findPolicies(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/plugins/policies", args, ListPoliciesResponse.class).policies();
+  }
+
+  private <T> T doHttpGet(String path, Map<String, String> params, Class<T> bodyClass)
+      throws RangerException {
+    URI uri;
+    try {
+      uri =
+          new URIBuilder(baseUri + path)
+              .addParameters(
+                  params.entrySet().stream()
+                      .map(entry -> new BasicNameValuePair(entry.getKey(), entry.getValue()))
+                      .collect(toImmutableList()))
+              .build();
+    } catch (URISyntaxException e) {
+      throw new RangerException("Failed to build API URI", e);
+    }
+    HttpGet httpRequest = new HttpGet(uri);
+    httpRequest.addHeader(ACCEPT, APPLICATION_JSON.toString());
+    try {
+      httpRequest.addHeader(new BasicScheme().authenticate(credentials, httpRequest));
+    } catch (AuthenticationException e) {
+      throw new RangerException("Failed to initialize authentication header", e);
+    }
+    CloseableHttpResponse httpResponse = null;
```

Comment:
Should this be moved in a `t-w-r` to ensure it will get closed?

Reply:
Good catch, this would have definitely leaked connections.